sbi/samplers/rejection/rejection.py

```diff
@@ -188,9 +187,10 @@ def log_prob(self, theta: Tensor, **kwargs) -> Tensor:
 
 @torch.no_grad()
 def accept_reject_sample(
-    proposal: Union[nn.Module, Distribution],
+    proposal: Callable,
```

Comment:
not sure whether we want to do this as part of this PR, but an idea: we could define a Python protocol that has a __call__ methods that takes a `Shape` as input arg (plus `**kwargs`). All methods that share this signature will then follow this protocol and the type checker will not complain I think.

Reply:
Interesting! I think it will help in making sure we write consistent `sample()` methods for different classes. I am a bit worried that this adds another level of abstraction for users. Currently, to produce samples, we use `accept_reject_sample()`, which takes a `proposal` from a `Posterior` class, which is generated using a `Potential` object calling an `estimator`. I would propose we try to simplify, rather than adding more objects here 😅

Reply:
move to issue: https://github.com/sbi-dev/sbi/issues/1395